src/pykiso/lib/connectors/cc_process.py

```diff
@@ -0,0 +1,319 @@
+            encoding=self.encoding,
+            cwd=self.cwd,
+            env=self.env,
+        )
+
+        if self.pipe_stdout:
+            self.stdout_thread = self._start_read_thread(self.process.stdout, "stdout")
+        if self.pipe_stderr:
+            self.stderr_thread = self._start_read_thread(self.process.stderr, "stderr")
+
+    def _start_read_thread(self, stream: IO, name: str) -> threading.Thread:
+        """Start a read thread
+
+        :param stream: The stream to read from
+        :param name: The name of the stream
+
+        :return: The thread object
+        """
+        thread = threading.Thread(
+            name=f"cc_process_{name}", target=self._read_thread, args=(stream, name)
+        )
+        thread.start()
+        return thread
+
+    def _read_thread(self, stream: IO, name: str) -> None:
+        """Thread for reading data from stdout or stderr
+
+        :param stream: The stream to read from
+        :param name: The name of the stream
+        """
+        try:
+            while True:
+                if self.text:
+                    data = stream.readline()
+                else:
+                    data = stream.read(1)
+                if len(data) == 0:
+                    break
+                self.queue_in.put(ProcessMessage(name, data))
+        finally:
+            with self.lock:
+                self.ready += 1
+                if self.ready == int(self.pipe_stdout) + int(self.pipe_stderr):
+                    # PROCESS_EXIT marks the termination of all read threads
+                    self.queue_in.put(PROCESS_EXIT)
+
+    def _cc_close(self) -> None:
+        """Close the channel."""
+        self._cleanup()
+
+    def _cc_send(self, msg: MessageType, raw: bool = False, **kwargs) -> None:
+        """Execute process commands or write data to stdin
+
+        :param msg: data to send
+        :param raw: unused
+
+        :raises CCProcessError: Stdin pipe is not enabled
+
+        """
+        if isinstance(msg, dict) and msg.get("command") == "start":
+            self.start(msg.get("executable"), msg.get("args"))
+        elif self.pipe_stdin:
+            if self.process is None:
+                raise CCProcessError("Process is not running.")
+            log.internal_debug(f"write stdin: {msg}")
+            self.process.stdin.write(msg)
+            self.process.stdin.flush()
+        else:
+            raise CCProcessError("Can not send to stdin because pipe is not enabled.")
+
+    def _cleanup(self) -> None:
+        """Cleanup threads and process objects"""
+        if self.process is not None:
+            # Terminate the process if still running
+            self.process.terminate()
+            try:
+                self.process.wait(5)
+            except subprocess.TimeoutExpired:
+                log.internal_warning(
+                    f"Process {self.executable} could not be terminated"
+                )
+                self.process.kill()
```

Comment:
On windows, kill does exactly the same as terminate so I don't think this will be useful

Reply:
Will not change that as it also does not cause a problem.